### PR TITLE
option: abbreviate common prefix

### PR DESCRIPTION
**Pull request by MarcelCoding**
image

### CODE REVIEW

File: src/app/core/components/search/search.component.ts [hunk 1]
```diff
@@ -36,6 +50,15 @@ export class SearchComponent implements OnInit, AfterViewInit, OnDestroy {
       formValue.scope === null ? undefined : formValue.scope,
       formValue.query ?? ''
     )),
+    map(options => {
+      const commonPrefix0 = prefix(options);
+      const idx = commonPrefix0.lastIndexOf('.');
+      const commonPrefix = commonPrefix0.substring(0, idx).split(".");
```
Comment (SuperSandro2000) [inline src/app/core/components/search/search.component.ts:56]: Why the extra 0 at commonPrefix?

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
wait a second...

**Comment by SuperSandro2000** [comment]
``<name>`` is not shortening that nice...


image


Can we shorten only if the option would be to long?

image

The red ones should probably not be shortened IMO

Also should we replace the p.n.c.c with a elipse to better indicate it is shortened?

**Comment by felbinger** [comment]
> elipse

*ellipsis

**Comment by SuperSandro2000** [comment]
After a big discussing we found a better idea:
we add a checkbox to open the search panel with double width

**Comment by SuperSandro2000** [comment]
Now I somehow don't get any results 😅

**Comment by SuperSandro2000** [comment]
This PR is cursed, where is the home-manager prefix?!

https://12456131.nuschtos-search.pages.dev/?option=accounts.calendar.accounts.%3Cname%3E.remote.url&query=calendar

**Comment by MarcelCoding** [comment]
...

**Pull request merged**